transaction-view/src/address_table_lookup_meta.rs

```diff
@@ -0,0 +1,167 @@
+use {
+    crate::{
+        bytes::{
+            advance_offset_for_array, advance_offset_for_type, check_remaining,
+            optimized_read_compressed_u16, read_byte,
+        },
+        result::Result,
+    },
+    solana_sdk::{packet::PACKET_DATA_SIZE, pubkey::Pubkey},
+};
+
+/// Contains meta-data about the address table lookups in a transaction packet.
+pub struct AddressTableLookupMeta {
+    /// The number of address table lookups in the transaction.
+    pub(crate) num_address_table_lookup: u16,
+    /// The offset to the first address table lookup in the transaction.
+    pub(crate) offset: u16,
+}
+
+impl AddressTableLookupMeta {
+    /// Get the number of address table lookups (ATL) and offset to the first.
+    /// The offset will be updated to point to the first byte after the last
+    /// ATL.
+    /// This function will parse each ATL to ensure the data is well-formed,
+    /// but will not cache data related to these ATLs.
+    pub fn try_new(bytes: &[u8], offset: &mut usize) -> Result<Self> {
+        // Read the number of ATLs at the current offset.
+        // Each ATL needs at least 34 bytes, so do a sanity check here to
+        // ensure we have enough bytes to read the number of ATLs.
+        const MIN_SIZED_ATL: usize = core::mem::size_of::<Pubkey>() + 1 + 1;
+        const MAX_ATLS_PER_PACKET: usize = PACKET_DATA_SIZE / MIN_SIZED_ATL;
+        // Maximum number of ATLs should be represented by a single byte,
+        // thus the MSB should not be set.
+        const _: () = assert!(MAX_ATLS_PER_PACKET & 0b1000_0000 == 0);
+        let num_address_table_lookups = u16::from(read_byte(bytes, offset)?);
```

Comment:
The number is a `u8`. Why did you choose `u16` to save it?

Reply:
was consistently using u16s for all my sizes; but you're right. I should just use u8s whenever those cannot validly exceed a u8.

Reply:
c0d298dc65c7fb06d5d58caca17fc32467da77dc

Reply:
After my 2 open PRs get merged I'll post PRs to update the other u16s that could be u8s.

Off top of my head these are: `num_signatures` and `num_static_accounts`